lib/polly_stubs.c

```diff
@@ -119,4 +119,39 @@ caml_polly_wait(value val_epfd, value val_max, value val_timeout, value val_f)
 	CAMLreturn(Val_int(ready));
 }
 
+CAMLprim value
+caml_polly_wait_fold(value val_epfd, value val_max, value val_timeout,
+		     value val_init, value val_f)
+{
+	CAMLparam5(val_epfd, val_max, val_timeout, val_init, val_f);
+	value args[4];
+
+	struct epoll_event *events;
+	int ready, i;
+
+	if (Int_val(val_max) <= 0)
+		uerror(__FUNCTION__, Nothing);
```

Comment:
uerror here will raise an exception with whatever errno happened to be set to (which could be 0, or someone else's errno). Better to use an explicit error here, 'caml_invalid_argument'  from 'fail.h' perhaps?

Reply:
Same thing in the older function above.